docs/docs/entities/entity-store.mdx

```diff
@@ -170,9 +170,13 @@ store.remove();
 Insert or update an entity. Creates a new entity when no entity matches the `id`; otherwise, it performs an update:
 
 ```ts
-store.upsert(id, { isOpen: true });
+store.upsert(id, { isOpen: true }, (id, newState) => ({ id, ...newState });
```

Comment:
Regarding the previous comment, we can make this object as default: `(id, newState) => ({ id, ...newState });`

Reply:
This would only be possible if typescript knew in advance that all properties of a state are optional: But this is not possible. So it is mandatory to specify an initialization callback which returns the new state with all required properties.  At least it is an identity function that passes the previously updated state through unmodified if all non-optional fields are already set.

Reply:
Don't you think it's annoying to do:
`store.upsert(id, { isOpen: true }, (id, newState) => ({ id, ...newState });` in each call?

Reply:
I totally agree with you that it is annoying, but I wouldn't know how to write this as a default parameter and at the same time respecting non-optional state properties that require a custom initialization callback.

Reply:
Why not make it optional and do something like:

```ts
if(!isFunction(onCreate)) {
  entity = { [this.idKey]: id }
}
```

Reply:
Because it would potentially produce incorrect entity states, if non optional properties are defined:
```ts
interface State {
   id: number;
   name: string;
}
```
Here, the `name` property would be `undefined` but according to its type definition it is never undefined. So typescript will not show a warning that the property is undefined and must be checked for `undefined` value before accessing it.

Reply:
Yes, I understand it. But for some developers, it's doesn't matter. If someone wants this feature, they'll pass the onCreate function. Otherwise, it will work the same as it works now.

Reply:
Ok, that's a point, but then there should be a warning in the docs which implications it has to omit the `onCreate` callback.

Another possible compromise would be to create a derived entity store class in which all state properties are automatically enforced to be optional. Then `onCreate` could always be omitted in `upsert`.

Reply:
Update the docs and mention it. Thanks.